### DISCUSSION TITLE
FR: Support `omnifunc` for the completion

### DISCUSSION THREAD

**Discussion by gh-liu** [olimorris/codecompanion.nvim#475] [General]
Thanks for your amazing work!
Maybe this plugin could support [omnifunc](https://vimdoc.sourceforge.net/htmldoc/options.html#'omnifunc') for the completion, which is a buffer local option be used for insert mode completion.

**Comment by olimorris** [comment]
Why do we need it after https://github.com/olimorris/codecompanion.nvim/commit/a1147f8133707251603525acf6a87f5c12ef3a70?

**Comment by gh-liu** [comment]
IMHO, its the vim way to provide completion in the insert mode, if this awsome plugin support `omnifunc`:
- first, we do not have to map one another key to the completion
- second, it will compatible well with other completion plugins, cmp already support `omnifunc` by [cmp-omni](https://github.com/hrsh7th/cmp-omni), and blink.cmp may support it in the [future](https://github.com/Saghen/blink.cmp/issues/330).

**Reply by olimorris** [reply]
If we're utilising native Neovim functions like `vim.fn.complete()` then that is the "Neovim way". We're not fighting the editor to add functionality.

I like the idea of your second point. However, it still requires users to add `cmp-omni` support, whereas the current implementation works ootb (albeit a user may have to change the mappings).
